updatecli/updatecli.d/rancher/shell/helm-project-operator/shell.yaml

```diff
@@ -0,0 +1,67 @@
+---
+name: "Bump rancher/shell image version"
+scms:
+  shell:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ .github.username }}"
+      token: "{{ requiredEnv .github.token }}"
+      owner: "rancher"
+      repository: "shell"
+      branch: "master"
+  helm-project-operator:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ .github.username }}"
+      token: "{{ requiredEnv .github.token }}"
+      owner: "rancher"
+      repository: "helm-project-operator"
+      branch: "main"
+
+pullrequests:
+  github:
+    kind: "github"
+    scmid: "helm-project-operator"
+    spec:
+      automerge: false
+
+sources:
+  shell:
+    name: "Get rancher/shell image latest or -rc version"
+    kind: "gittag"
+    scmid: "shell"
+    spec:
+      versionfilter:
+        kind: "latest"
+
+conditions:
+  shell:
+    name: "Check rancher/shell image version in DockerHub"
+    kind: "dockerimage"
```

Comment:
Any reason we use `githubrelease` for other components but `dockerimage` for shell? We already found an inconsistency in DockerHub and it makes sense to keep it uniform across components (unless we need to look at DockerHub). Also we don't use a token so we can get rate limited on DockerHub I guess?

Reply:
> Any reason we use githubrelease for other components but dockerimage for shell?
The condition here tests that a docker image named "rancher/shell:<latest_tag retrieved from source>" exists on Dockerhub.
> Also we don't use a token so we can get rate limited on DockerHub I guess?
And you are gonna be affected by rate limit if no credentials are provided.

The goal of this condition is to be sure that we don't update a Helm chart if for some reason we created a new release by publishing a git tag but for some reason it failed building the docker image associated to that specific tag

Reply:
This is the explanation ^ .

Reply:
> Also we don't use a token so we can get rate limited on DockerHub I guess?

Yes, this can happen eventually. As an improvement, I can request a read-only Dockerhub token to EIO.

Reply:
Issue to track this improvement - https://github.com/rancherlabs/updatecli-automation/issues/3.